Sources/WeaverDI/ThreadSafeInstanceCachingDecorator.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  ThreadSafeInstanceCachingDecorator.swift
+//  WeaverDI
+//
+//  Created by Bruno Mazzo on 5/31/18.
+//
+
+import Foundation
+
+class ThreadSafeInstanceCachingDecorator: InstanceCaching {
```

Comment:
Same here, let's make it `final` and rename it to `ThreadSafeInstanceCacheDecorator`

Reply:
Done